Source/santad/ProcessTree/annotations/originator.cc

```diff
@@ -0,0 +1,61 @@
+/// Copyright 2023 Google LLC
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     https://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+#include "Source/santad/ProcessTree/annotations/originator.h"
+
+#include <optional>
+
+#include "Source/santad/ProcessTree/process_tree.pb.h"
+#include "Source/santad/ProcessTree/process_tree.h"
+
+namespace ptpb = ::santa::pb::v1::process_tree;
+
+namespace santa::santad::process_tree {
+
+void OriginatorAnnotator::AnnotateFork(ProcessTree &tree, const Process &parent,
+                                       const Process &child) {
+  // "Base case". Propagate existing annotations down to descendants.
+  if (auto annotation = tree.GetAnnotation<OriginatorAnnotator>(parent)) {
+    tree.AnnotateProcess(child, std::move(*annotation));
+  }
+}
+
```

Comment:
Outside of scope for this PR, but I think this makes it clear in the future having access to codesign info would be very powerful for macOS annotators. Has there been any thought to how this information could be propagated, maybe in a `Process` object?

Reply:
Yep so the thought right now is that any non-OS-agnostic stuff (e.g. macOS codesigning) will be in an `agent_specific` or `os_specific` struct in `Process` (defined in the `process_tree` OS-specific headers) which anything can then get/set. This implementation will change slightly when the internal changes are migrated out (notably the fact that everything is a proto for serialization) but the idea remains the same.